src/components/interactive-map/interactive-map.tsx

```diff
@@ -303,12 +335,15 @@ const InteractiveMap: React.FC<InteractiveMapProps> = (props) => {
             />
           );
         })}
+
       {Object.entries(props.orders)
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
         .filter(([_, order]) => order.type === "move")
         .map(([provinceId, order]) => {
           const source = props.map.provinces.find((p) => p.id === provinceId);
           const target = props.map.provinces.find((p) => p.id === order.target);
+          const unit = props.units[provinceId];
+          const unitColor = unit ? props.nationColors[unit.nation] : "black"; // Default to black if no unit found
```

Comment:
I'll do a subsequent iteration to tidy up some of this repeated code and stuff. I think I can restructure this file a little more nicely.